### PR TITLE
Initialize lock module

### PR DESCRIPTION
**Pull request by JrSchild**
### Context
Create the initial lock module abstracted from our current projects.

### What has been done
- Added the enrise-lock mechanism, code, README docs, travis, tests, package.json, LICENSE, eslint, test-coverage, Travis and .gitignore

### How to test
- Run `npm run test`, all tests pass with 100% test coverage

### Notes
- The buttons in the README will start working after the merge to master

### CODE REVIEW

**Comment by JrSchild** [comment]
One sec, gotta add coveralls and Travis

**Comment by JrSchild** [comment]
Okay this PR is ready. Also removed the option for coveralls to comment the test coverage.

File: README.md [hunk 1]
```diff
@@ -1 +1,60 @@
 # Node.js lock module
+
+[![build:?](https://img.shields.io/travis/Enrise/node-lock.svg?style=flat-square)](https://travis-ci.org/Enrise/node-lock)
+[![Coverage Status](https://img.shields.io/coveralls/Enrise/node-lock/master.svg?style=flat-square)](https://coveralls.io/github/Enrise/node-lock?branch=master)
+[![dependencies:?](https://img.shields.io/david/Enrise/node-lock.svg?style=flat-square)](https://david-dm.org/Enrise/node-lock)
+[![devDependencies:?](https://img.shields.io/david/dev/Enrise/node-lock.svg?style=flat-square)](https://david-dm.org/Enrise/node-lock)
+
+> An elasticsearch based lock mechanism.
```
Comment (roelandvanbatenburg) [inline README.md]: Can you expand on this a bit? Why is this useful? In what type of environment. What are these owner and resource concepts?

File: index.js [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+  if (!(_.isString(resource) && _.isString(owner) && _.isFunction(cb) && resource.length && owner.length && cb)) {
+    throw new Error('Incorrect parameter(s)');
+  }
+}
+
+class Lock {
+  constructor(settings) {
+    this.esClient = settings.esClient;
+    this.index = settings.index;
+    this.type = settings.type;
```
Comment (roelandvanbatenburg) [inline index.js]: I think this can use a default (change also the README)

**Comment by JrSchild** [comment]
🇺🇸

**Pull request merged**